src/SQLStore/SimpleListLookup.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace SMW\SQLStore;
+
+/**
+ * A basic interface for fetching a simple list either from a DB or cache
+ * instance
+ *
+ * @license GNU GPL v2+
+ * @since 2.2
+ *
+ * @author mwjames
+ */
+interface SimpleListLookup {
```

Comment:
> It has a single method inherently relevant to list lookup stuff. The other 3 make sense in caching context. I do not think they belong to the base interface. For what reasons did you put them there?

I thought about it and the way I envisioned this was that `SimpleListLookup` contains all methods that are necessary the communicate with a `SimpleListLookup` instance whether it is a concrete class that does the lookup or a caching layer (such as `ListLookupCache`)  but one is indifferent to the concrete implementation detail.

Knowing that ` ... instance of SimpleListLookup` you have a contract to four methods without the need for a consumer to know that is coming from cache or a DB lookup (well you know it indirectly after the execution and not before by checking on `isCached`).

If `SimpleListLookup` has only the `fetchResultList` method then one has always to check that it is `SimpleListLookup` or a concrete `ListLookupCache` instance because `isCached` is`getTimestamp` is only provided by `ListLookupCache`.